docusaurus/docs/Android/02-client/06-guides/03-channel-list-updates.mdx

```diff
@@ -0,0 +1,124 @@
+# Channel List Updates
+
+The `ChannelList` component is being updated based on the data from two sources:
+- Query channels request, with given filter and query sort, when the component is initialized
+- Channel-related events that come from the `WebSocket` when the socket is connected
+
+Unlike query channel requests, where you can filter particular channels, channel-related events are not being filtered and need additional attention to handle the list updates correctly.
+By default, the SDK updates the list based on membership - the channel will be added if the currentUser is a member and will be removed otherwise.
+
+## Custom ChatEventHandler
+
+The default event handling logic might not fit into your use case and you can spot that `ChannelListView` is not updated properly.  This means you should provide a custom `ChatEventHandler`.
+You can choose between overriding:
+- `DefaultChatEventHandler` - when you need to extend member-based behavior by adding additional logic
+- `BaseChatEventHandler` - when you need to support different use cases, for example, non-member based but still a separation between `CidEvent` and `HasChannel` events is useful
+- `ChatEventHandler` - if you want to have full control over events' types
+
+In each case, you need to decide what to do with the particular event by returning `EventHandlingResult` that matches an action.
+You can find more details about the result class [here](https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/event/handler/chat/ChatEventHandler.kt#L57).
+
+:::note
+The `ChatEventHandler` is being provided by `ChatEventHandlerFactory` which gives you access to the visible channels map. By using the map, you can skip unwanted channel list updates.
+:::
+
+Both XML and Compose `ChannelListViewModel` and `ChannelListViewModelFactory` allow you to pass `chatEventHandlerFactory` via constructor.
+Alternatively, you can pass custom handler to `ChatClient::queryChannelsAsState` function, if you are not using UI SDKs.
+
+## Multiple ChannelListView
+
+If your application contains more than one `ChannelListView`, each of them should implement its own, custom `ChatEventHandler`.
+Let's say we want to have two `ChannelListView`:
+- First one displaying `public` channels
+- Second one for displaying `private` ones
```

Comment:
we don't need full stops at the end here, because we're giving out examples, not statements